src/main/java/htsjdk/samtools/SamReader.java

```diff
@@ -381,7 +381,7 @@ public PrimitiveSamReaderToSamReaderAdapter(final PrimitiveSamReader p, final Sa
             this.resource = resource;
         }
 
-        PrimitiveSamReader underlyingReader() {
+        public PrimitiveSamReader underlyingReader() {
```

Comment:
`PrimitiveSamReaderToSamReaderAdapter` is not public - but even if it was, the underlying reader is still needed to get the `BAMFileReader` so its newly-exposed `getFileSpan()` and `createIndexIterator()` methods can be called.